dcmdump/src/main.rs

```diff
@@ -85,15 +89,20 @@ where
 }
 
 #[cfg(windows)]
-fn os_compatibility() {
-    control::set_virtual_terminal(true);
+fn os_compatibility() -> Result<(), ()> {
+    control::set_virtual_terminal(true)
 }
 
 #[cfg(not(windows))]
-fn os_compatibility() {}
+fn os_compatibility() -> Result<(), ()> {
+    Ok(())
+}
 
 fn main() {
-    os_compatibility();
+    os_compatibility().unwrap_or_else(|_| {
+        println!("Error setting OS compatibility");
+        std::process::exit(ERROR_COMPAT);
```

Comment:
Instead of exiting, could the application continue running without colored output?